src/test/scala/dpla/ingestion3/harvesters/oai/DefaultSourceTest.scala

```diff
@@ -0,0 +1,73 @@
+package dpla.ingestion3.harvesters.oai
+
+import org.scalatest._
+
+/**
+  * Tests for DefaultSourceTest
+  */
+class DefaultSourceTest extends FlatSpec {
+  val source = new DefaultSource
+  val endpoint = "http://repox.example.edu:8080/repox/OAIHandler"
+  val verb = "ListRecords"
+  val list = "manuscripts, newspapers, maps"
+  val prefix = "mods"
+  val token = "1234"
+
+  "getEndpoint" should "throw Exception if endpoint missing" in {
+    val params = Map("foo" -> "bar")
+    assertThrows[Exception](source.getEndpoint(params))
+  }
+  it should "return endpoint as String" in {
+    val params = Map("path" -> endpoint)
+    assert(source.getEndpoint(params) === endpoint)
+  }
+
+  "getVerb" should "throw Exception if verb missing" in {
+    val params = Map("foo" -> "bar")
+    assertThrows[Exception](source.getVerb(params))
+  }
+  it should "return endpoint as String" in {
+    val params = Map("verb" -> verb)
+    assert(source.getVerb(params) === verb)
+  }
+
+  "getHarvestAllRecords" should "return false if harvestAllSets missing" in {
+    val params = Map("foo" -> "bar")
+    assert(source.getHarvestAllSets(params) === false)
+  }
+  it should "return true if harvestAllSets is true" in {
+    val params = Map("harvestAllSets" -> "true")
+    assert(source.getHarvestAllSets(params) === true)
+  }
+  it should "return false if harvestAllSets is false" in {
+    val params = Map("harvestAllSets" -> "false")
+    assert(source.getHarvestAllSets(params) === false)
+  }
+  it should "ignore case" in {
+    val params = Map("harvestAllSets" -> "TRUE")
+    assert(source.getHarvestAllSets(params) === true)
+  }
+  it should "throw Exception if harvestAllSets is neither true nor false" in {
+    val params = Map("harvestAllSets" -> "foo")
+    assertThrows[Exception](source.getHarvestAllSets(params))
+  }
+
+  "getSetlist" should "return None if setlist missing" in {
+    val params = Map("foo" -> "bar")
+    assert(source.getSetlist(params) === None)
+  }
+  it should "return set list as Some[Array[String]]" in {
+    val params = Map("setlist" -> list)
+    // TODO: Is there a better way to match Some[Array[String]]?
+    assert(source.getSetlist(params).get.toSeq === Array("manuscripts", "newspapers", "maps").toSeq)
```

Comment:
This doesn't need the additional transformation `.toSeq` just calling get() is enough for a valid comparison but I am confused why you can't compare successfully compare two options. The values are identical but `getSetList(params) === Some(Array(...))` fails